quantization/existing-data.ipynb

```diff
@@ -0,0 +1,199 @@
+    "mongo_client = pymongo.MongoClient(\"<ATLAS-CONNECTION-STRING>\")\n",
+    "db = mongo_client[\"sample_airbnb\"]\n",
+    "collection = db[\"listingsAndReviews\"]\n",
+    "\n",
+    "# Filter to exclude null or empty summary fields\n",
+    "filter = { \"summary\": {\"$nin\": [None, \"\"]} }\n",
+    "\n",
+    "# Get a subset of documents in the collection\n",
+    "documents = collection.find(filter).limit(50)\n",
+    "\n",
+    "# Initialize the count of updated documents\n",
+    "updated_doc_count = 0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for doc in documents:\n",
+    "    # Generate embeddings based on the summary\n",
+    "    summary = doc[\"summary\"]\n",
+    "    embedding = get_embedding(summary)  # Get float32 embedding\n",
+    "\n",
+    "    # Convert float32 embeddings into BSON format\n",
+    "    bson_vector = generate_bson_vector(embedding, BinaryVectorDtype.FLOAT32)\n",
+    "\n",
+    "    # Uncomment one of following lines if you specified a different data type in the get_embedding function\n",
+    "    # bson_vector = generate_bson_vector(embedding, BinaryVectorDtype.INT8)\n",
+    "    # bson_vector = generate_bson_vector(embedding, BinaryVectorDtype.PACKED_BIT) # int1\n",
+    "\n",
+    "    # Update the document with the BSON embedding\n",
+    "    collection.update_one(\n",
+    "        {\"_id\": doc[\"_id\"]},\n",
+    "        {\"$set\": {\"embedding\": bson_vector}}\n",
+    "    )\n",
+    "    updated_doc_count += 1\n",
+    "\n",
+    "print(f\"Updated {updated_doc_count} documents with BSON embeddings.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pymongo.operations import SearchIndexModel\n",
+    "\n",
+    "# Define and create the vector search index\n",
+    "vector_search_index_definition = {\n",
+    "  \"fields\":[\n",
+    "    {\n",
+    "      \"type\": \"vector\",\n",
+    "      \"path\": \"embedding\",\n",
+    "      \"similarity\": \"euclidean\",\n",
+    "      \"numDimensions\": 1024\n",
+    "    }\n",
+    "  ]\n",
+    "}\n",
+    "\n",
+    "search_index_model = SearchIndexModel(definition=vector_search_index_definition, name=\"<INDEX-NAME>\", type=\"vectorSearch\")\n",
+    "\n",
+    "collection.create_search_index(model=search_index_model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define function to run a vector search query\n",
+    "def run_vector_search(query_text, collection, path):\n",
+    "  query_embedding = get_embedding(\"query_text\")\n",
+    "  bson_query_vector = generate_bson_vector(query_embedding, BinaryVectorDtype.FLOAT32)\n",
+    "\n",
+    "  pipeline = [\n",
+    "    {\n",
+    "      '$vectorSearch': {\n",
+    "        'index': '<INDEX-NAME>',\n",
```

Comment:
Same comment as the other example - if we're asking the user to supply the same value to populate the placeholder in more than one place - `<INDEX-NAME>` - we should factor it out to a variable we're using in multiple places.